pkg/bloomgateway/client.go

```diff
@@ -276,10 +304,12 @@ func (c *GatewayClient) serverAddrsForFingerprints(tenantID string, groups []*lo
 
 	fingerprints := make([]uint64, numFingerprints)
 	addresses := make([][]string, numFingerprints)
-	bufDescs, bufHosts, bufZones := ring.MakeBuffersForGet()
+
+	buf := ringGetBuffersPool.Get().(*ringGetBuffers)
+	defer ringGetBuffersPool.Put(buf)
```

Comment:
Should we clear the `buf.Descs`, `buf.Hosts`, and `buf.Zones` before releasing them back to the pool? It would release the items in the slices earlier for GC, right?

Reply:
yep, I believe we clean up the slice only when we fetch it from the pool, so, we will keep the objects in memory until the next time we work with this slice. counting that we have the slices with different sizes, some slices might be rarely used and they will keep some memory.

Reply:
Good idea, done.